hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/fs/viewfs/TestViewFileSystemHdfs.java

```diff
@@ -479,4 +485,22 @@ public Object run() throws IOException {
     assertEquals("The owner did not match ", owner, userUgi.getShortUserName());
     otherfs.delete(user1Path, false);
   }
+
+  @Test
```

Comment:
This test does not seem to reproduce the problem. If I revert the change it still passes. 
I believe permission = null is replaced by the default permission somewhere down in the impl.

Reply:
You are right.. Looks like the permissions are inconsistent. With the change the default perm we use in `mkdirs` is 777 but `fs.getFileStatus(xxx).getPermission()` returns 751 for both internal and real dirs (regardless of whether we revert the change or not). I'm not sure why but I'll dig into this.

Reply:
Removed the override method, kept the test because the test itself makes sense, the test passes without any change because we already handled it in downstream impl DFSClient#mkdirs.